py/picca/cf.py

```diff
@@ -155,9 +155,16 @@ def compute_xi(healpixs):
 
             for delta2 in delta1.neighbours:
                 ang = delta1.get_angle_between(delta2)
-                same_half_plate = ((delta1.plate == delta2.plate) and (
-                    (delta1.fiberid <= 500 and delta2.fiberid <= 500) or
-                    (delta1.fiberid > 500 and delta2.fiberid > 500)))
+                if remove_same_half_plate_close_pairs:
```

Comment:
This is a new addition that is required for the deltas produced with the delta_extraction code to work. The reason behind this need is that now plate, fiberid, and mjd (when reading SDSS data) are strings including the information from the different reobserations, and this was breaking this check. 

The `same_half_plate variable` is only used when `remove_same_half_plate_close_pairs` is set to `True`. This is an option that was used when no coadding of the forests was applied to correct for correlated sky noise resiudals. It does not make sense for it to run when reobservations are coadded so it makes sense that the code crashes then. It would probably be better to improve the error reporting, but I think that we should just remove this option (or at least deprecate it)

Answering your question, `remove_same_half_plate_close_pairs` is a module variable that is set in `picca_cf.py` that is why there are no changes prior to this. It was used in the functions `compute_xi_forest_pairs` and the equivalent for the dmat and the metal dmat

Reply:
ah, that should be ok, maybe add a deprecation warning when this option is used, so that people get a note if they still use it....